### ISSUE TITLE
Very low Loss values

### ISSUE THREAD

**Issue by Manu752** [NVlabs/Deep_Object_Pose#408]
Hi i am trying to use DOPE to pick up a cube and I have trained it using synthetic data using blender .When I start to train my loss values are very low from the begining itself . Could someone please assist .

![Image](https://github.com/user-attachments/assets/988cf2f7-be9a-4eac-998c-901fc47612dd)

The command  :~/dope_repo_src/Deep_Object_Pose/train#   python3 -m torch.distributed.launch --nproc_per_node=1 train.py \
>     --data /output \
>     --object cube \
>     --batchsize 16 \
>     --outf /trained_models \
>     --epochs 5 
/root/dope_repo_src/Deep_Object_Pose/dope_env/dope_env/lib/python3.8/site-packages/torch/distributed/launch.py:208: FutureWarning: The module torch.distributed.launch is deprecated
and will be removed in future. Use torchrun.
Note that --use-env is set by default in torchrun.
If your script expects `--local-rank` argument to be set, please
change it to read from `os.environ['LOCAL_RANK']` instead. See 
https://pytorch.org/docs/stable/distributed.html#launch-utility for 
further instructions

  main()
Number of Training Images: 6005
training data: 376 batches
Loading Model...
ready to train!
start: 06/04/2025, 12:54:33
NaN or Inf found in input tensor.
Train Epoch: 0 [0/6005 (0%)] 	Loss: 0.035927392542362 	Local Rank: 0
Train Epoch: 0 [1600/6005 (27%)] 	Loss: 0.001287658000365 	Local Rank: 0
Train Epoch: 0 [3200/6005 (53%)] 	Loss: 0.000217389111640 	Local Rank: 0
Train Epoch: 0 [4800/6005 (80%)] 	Loss: 0.000109579552372 	Local Rank: 0
NaN or Inf found in input tensor.
Train Epoch: 1 [0/6005 (0%)] 	Loss: 0.000067033462983 	Local Rank: 0
Train Epoch: 1 [1600/6005 (27%)] 	Loss: 0.000041542443796 	Local Rank: 0
Train Epoch: 1 [3200/6005 (53%)] 	Loss: 0.000036422810808 	Local Rank: 0
Train Epoch: 1 [4800/6005 (80%)] 	Loss: 0.000023922695618 	Local Rank: 0
NaN or Inf found in input tensor.

**Comment by Manu752** [comment]
It is getting more pixelated ..I am now using the weights from the 300th epoch and this is how it looks --

![Image](https://github.com/user-attachments/assets/af960d44-0b13-4121-a46a-a0981ce6aaa8)

**Comment by nv-jeff** [comment]
This could be a problem with your data, or it could just be slow convergence.  I would recommend two approaches:
1. Keep training.  I found that ~2000 epochs with 50k images were required to give "as good as the paper" results.  With 35k images, you may need more.
2. Try training on just one image.  If you train on one image for, say, 10k epochs or so, does it overfit and converge (i.e. regurgitate the training data)?

**Comment by Manu752** [comment]
Hi @nv-jeff  , I am not sure how my data is corrupt . The  " ground truth " appears to be correct I can see the the cuboid . I am not sure why the model is unable to learn a simple cube . I will proceed with option 2 first .

**Comment by Manu752** [comment]
Hi again @nv-jeff   ,  So I tried option 2 and It isnt working :( . I triend on a single image for 30,000 epochs and this is how the beief map looks like for the same image . I am now not sure where I am wrong -

![Image](https://github.com/user-attachments/assets/a959d880-69c6-45c9-9f86-827e8ca49fbb)

**Comment by Manu752** [comment]
The algorithm works perfectly when I use predefiend weights (Yogurt.pth for exampole)-

![Image](https://github.com/user-attachments/assets/35cbd2cd-fbee-40a3-b747-21cac43e8bb9)

**Comment by nv-jeff** [comment]
Can you share your JSON file for the training image you show above?

**Comment by Manu752** [comment]
Thanks for the prompt reply  . Here is the json that the  Blendor script generated --
{
    "camera_data": {
        "width": 680,
        "height": 480,
        "camera_look_at": {
            "at": [
                -0.0,
                1.0,
                -7.549790126404332e-08
            ],
            "eye": [
                -0.0,
                25.0,
                -0.0
            ],
            "up": [
                1.0,
                0.0,
                0.0
            ]
        },
        "intrinsics": {
            "fx": 820.8327696058485,
            "fy": 820.8327696058485,
            "cx": 339.5,
            "cy": 239.5
        }
    },
    "objects": [
        {
            "class": "models",
            "name": "models_000",
            "visibility": 2274,
            "projected_cuboid": [
                [
                    446.0798217040395,
                    346.9202202791824
                ],
                [
                    462.06783478801054,
                    370.6674946404793
                ],
                [
                    486.3597503189185,
                    344.14135108123867
                ],
                [
                    471.1389994055704,
                    320.04013271299107
                ],
                [
                    422.69098580407433,
                    334.36591547124243
                ],
                [
                    438.6340962513455,
                    357.5424284525774
                ],
                [
                    462.3192435598466,
                    332.1070517839031
                ],
                [
                    447.10094472365824,
                    308.60810344048906
                ],
                [
                    454.4904482734505,
                    339.2482721040143
                ]
            ],
            "location": [
                15.832285394299683,
                -13.733676331901913,
                -113.01511180890823
            ],
            "quaternion_xyzw": [
                0.3013881262927946,
                0.006267151522942752,
                0.8640301124323593,
                0.40320948023531133
            ]
        }
    ]
}

**Comment by Manu752** [comment]
Ground truth --

![Image](https://github.com/user-attachments/assets/d629b8fa-f08b-456f-b49c-bd3ba7b03a68)

**Comment by Manu752** [comment]
I live in Germany ..So I am off to bed right now . I will check this tomorrow first thing :)

**Comment by nv-jeff** [comment]
I see your problem. In the JSON file, your object is named `models` but on the command line, you're asking DOPE to train on the location of objects named "Cube."  So, no training is happening.  

I'm guessing you didn't specify an `--object_class` when you executed the BlenderProc script, so it used the name of the containing directory.  

The easiest fix for this is to replace all the occurrences of `"class": "models"` in your JSON files with '"class": "cube"`

**Comment by nv-jeff** [comment]
Let us know if this solves your problem.

I have some small changes I will check in soon that should prevent this kind of silent failure from happening in the futrue. (For example, requiring and explicit class name during data generation, and raising an error during training if there are no classes in the training data that match the command-line parameter)

**Comment by Manu752** [comment]
Forgot to mention I run this in a docker container and when I run nvidia-smi  I am at 100% capacity  -|   0  NVIDIA RTX A4500               Off |   00000000:65:00.0  On |                  Off |
| 51%   76C    P2            197W /  200W |   16790MiB /  20470MiB |    100%

**Comment by nv-jeff** [comment]
I am going to close this issue, as I believe we have diagnosed your dataset problem.  Could you please open a new one to discuss the speed issue?
